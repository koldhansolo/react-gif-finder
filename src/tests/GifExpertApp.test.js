import React from 'react';
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";


describe('Pruebas a <GifExpertApp />', () => {
	test('Hacer match con el snapshot', () => {
		const wrapper = shallow( <GifExpertApp /> );

		expect( wrapper ).toMatchSnapshot();
	});

	test('Debe de mostrar una lista de categorias', () => {
		// Defino dos categorias que serán pasadas por props
		const categories = [ 'One Punch', 'Dragon Ball' ];

		// Obtengo el componente GifExpertApp con las dos categorias que definí
		const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );

		expect( wrapper ).toMatchSnapshot();
		expect( wrapper.find( 'GifGrid' ).length ).toBe( categories.length );
	});
	
	
});
