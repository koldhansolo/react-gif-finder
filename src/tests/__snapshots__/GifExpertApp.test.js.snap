// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="One Punch"
    key="One Punch"
  />
  <GifGrid
    category="Dragon Ball"
    key="Dragon Ball"
  />
</div>
`;

exports[`Pruebas a <GifExpertApp /> Hacer match con el snapshot 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</div>
`;
